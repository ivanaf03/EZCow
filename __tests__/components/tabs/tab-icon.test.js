import React from "react";

import {render} from "@testing-library/react-native";

import TabIcon from "../../../components/tabs/tab-icon";

jest.mock("@fortawesome/react-native-fontawesome", () => ({
    FontAwesomeIcon: () => null
}));

describe("TabIcon", () => {

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("should render correctly", () => {
        const tree = render(<TabIcon name="Test" color="c_white" icon={null} size={24} />);
        expect(tree).toMatchSnapshot();
    });

    it("should render when is focused", () => {
        const tree = render(<TabIcon name="Test" color="c_white" icon={null} size={24} focused={true} />);
        expect(tree).toMatchSnapshot();
    });
});