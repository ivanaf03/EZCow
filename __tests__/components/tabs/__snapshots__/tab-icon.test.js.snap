// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabIcon should render correctly 1`] = `
<View
  style={
    [
      {
        "display": "flex",
      },
      {
        "marginLeft": -4,
        "marginTop": -4,
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "minWidth": 40,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Nunito-Medium",
        },
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 12,
          "lineHeight": 16,
        },
        [
          {
            "marginLeft": 4,
            "marginTop": 4,
          },
        ],
      ]
    }
  >
    Test
  </Text>
</View>
`;

exports[`TabIcon should render when is focused 1`] = `
<View
  style={
    [
      {
        "display": "flex",
      },
      {
        "marginLeft": -4,
        "marginTop": -4,
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "minWidth": 40,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Nunito-Bold",
        },
        {
          "color": "#6bc0eb",
        },
        {
          "fontSize": 12,
          "lineHeight": 16,
        },
        [
          {
            "marginLeft": 4,
            "marginTop": 4,
          },
        ],
      ]
    }
  >
    Test
  </Text>
</View>
`;
